### PR TITLE
feat: fallback to filename if recipe name is missing

### PR DESCRIPTION
**Pull request by ishanarya0**
When using the `gen` command, if `name` is not mentioned in `data`, it will fall back to the `FileName` as the recipe's name.

### CODE REVIEW

**Review by StewartJingga — CHANGES_REQUESTED** [review]
The related issue is not only specific to templating but when reading recipe from a file as well.

We need to:
- Fallback to file's name if name is not specified
- Strip the file extension as the recipe name on fallback

Also need help to also write tests for all those changes 🙏 .

**Comment by ishanarya0** [comment]
> The related issue is not only specific to templating but when reading recipe from a file as well.
> 
> We need to:
> 
> * Fallback to file's name if name is not specified
> * Strip the file extension as the recipe name on fallback
> 
> Also need help to also write tests for all those changes 🙏 .

Changes done.

File: recipe/generator_test.go [hunk 1]
```diff
@@ -14,41 +14,74 @@ func TestFromTemplate(t *testing.T) {
...
-				Data: map[string]interface{}{
-					"broker": "main-broker.com:9092",
-					"owner":  "john@example.com",
+
+		t.Run("when recipe has a name", func(t *testing.T) {
+			data := []recipe.TemplateData{
+				{
+					FileName: "recipe-one",
+					Data: map[string]interface{}{
+						"name":   "recipe-1",
```
Comment (mabdh) [inline recipe/generator_test.go:23]: would it be possible to source these data from the file?
like we do in normal cli?

Comment (ishanarya0) [reply recipe/generator_test.go]: Yes, did this. Thanks for the suggestion!

**Review by mabdh — APPROVED** [review]

**Pull request merged**